source/robot_model/include/robot_model/Model.hpp

```diff
@@ -109,38 +121,48 @@ class Model {
   void init_model();
 
   /**
-   * @brief Compute the jacobian from a given joint state at the joint level given in parameter
+   * @brief Compute the jacobian from a given joint state at the frame in parameter
    * @param joint_state containing the joint values of the robot
-   * @param joint_id id of the joint at which to compute the jacobian
+   * @param joint_id id of the frame at which to compute the jacobian
    * @return the jacobian matrix
    */
-  const StateRepresentation::Jacobian compute_jacobian(const StateRepresentation::JointState& joint_state, unsigned int joint_id);
+  StateRepresentation::Jacobian compute_jacobian(const StateRepresentation::JointState& joint_state, int frame_id);
```

Comment:
CLion told me that I could drop all the `const` in the function declaration. Is this wrong or do you have preferences?

Reply:
Okay that does make sense thanks. What do you think @buschbapti

Reply:
For those functions I don't mind removing the const as indeed there is no real issues. Although I am not entirely sure usage 2 is something we would do.

In general, we need to distinguish, however, the functions that return a reference to a class variable (e.g. getters). It generally make sense to keep the const for those if we want to enforce usage of setters.

Reply:
Sounds like a good compromise!